common/app/model/dotcomrendering/DotcomRenderingDataModel.scala

```diff
@@ -452,7 +455,15 @@ object DotcomRenderingDataModel {
       twitterHandle = content.tags.contributors.headOption.flatMap(_.properties.twitterHandle),
     )
 
-    val shouldAddAffiliateLinks = DotcomRenderingUtils.shouldAddAffiliateLinks(content)
+    def hasAffiliateLinks(
+        blocks: Seq[APIBlock],
+    ): Boolean = {
+      blocks.exists(block => DotcomRenderingUtils.stringContainsAffiliateableLinks(block.bodyHtml))
+    }
+
+    val isInDisableAffiliateLinksTest = ActiveExperiments.isParticipating(DisableAffiliateLinks)
```

Comment:
Similar to above, if this check is not used as part of `DotcomRenderingDataModel`, maybe we can make it in the `AffiliateLinksCleaner`, reducing indirection and making the logic easier to trace.

Reply:
I think for this point and the point above the question is whether we do the check further up, or if we pass the request to the cleaner. Because the check for the server side test participation requires access to the request